### DISCUSSION TITLE
Build action successful, where is executable?

### DISCUSSION THREAD

**Discussion by JamesBremner** [community/community#30452] [Actions] [answered]
I can successfully run a build action

![image](https://user-images.githubusercontent.com/2046227/186019384-32263de6-504d-49e5-a3cc-0950cdc078e5.png)

Neat!  But where is the executable that was built?  I want to download and run it.

I thought it would be in the repository's bib folder, but that is empty.

I thought it might be an "action artifact", but there are none listed

![image](https://user-images.githubusercontent.com/2046227/186019850-fa25278d-91e7-4704-8789-e77fc11f85e5.png)

Where should I look next?

James

**Comment by airtower-luna** [comment, marked as answer]
> Neat! But where is the executable that was built? I want to download and run it.

On the VM that ran the job, which has been deleted as soon as the job was done.

> I thought it might be an "action artifact", but there are none listed

Artifact is the right idea. :slightly_smiling_face: What you need to know is that artifacts aren't created automatically, you have to create them in your workflow, usually using [`actions/upload-artifact`](https://github.com/actions/upload-artifact).

E.g. if you want just that EXE (but you should still check the docs!):

```yaml
- uses: actions/upload-artifact@v3
  with:
    name: my-artifact
    path: bin/in2out.exe
```

**Reply by JamesBremner** [reply]
Works. Thanks!
